In [33]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

In [27]:
d1= pd.read_csv('BTCUSDT-trades-2022-03-01.csv',header=None,names=["1","Traded Price", "Traded Size", "Traded Notional", "Timestamp","6","7"],engine='python')
d1=d1.drop(['1','6','7'],axis=1)
d1.head()

Traded Price  Traded Size  Traded Notional      Timestamp
0      43160.00      0.00056        24.169600  1646092800000
1      43160.01      0.00052        22.443205  1646092800003
2      43160.01      0.00182        78.551218  1646092800004
3      43160.01      0.00631       272.339663  1646092800005
4      43160.00      0.00229        98.836400  1646092800006

In [28]:
d2= pd.read_csv('BTCUSDT-trades-2022-03-02.csv',header=None,names=["1","Traded Price", "Traded Size", "Traded Notional", "Timestamp","6","7"],engine='python')
d2=d2.drop(['1','6','7'],axis=1)
d2.head()

Traded Price  Traded Size  Traded Notional      Timestamp
0      44421.20      0.00097        43.088564  1646179200000
1      44423.46      0.00154        68.412128  1646179200002
2      44423.46      0.00039        17.325149  1646179200003
3      44421.20      0.00328       145.701536  1646179200003
4      44423.46      0.00075        33.317595  1646179200005

In [29]:
d3= pd.read_csv('BTCUSDT-trades-2022-03-03.csv',header=None,names=["1","Traded Price", "Traded Size", "Traded Notional", "Timestamp","6","7"],engine='python')
d3=d3.drop(['1','6','7'],axis=1)
d3.head()

Traded Price  Traded Size  Traded Notional      Timestamp
0      43892.99      0.00158        69.350924  1646265600000
1      43892.99      0.00066        28.969373  1646265600000
2      43892.98      0.00136        59.694453  1646265600000
3      43892.99      0.00045        19.751846  1646265600000
4      43892.98      0.00048        21.068630  1646265600001

In [48]:
d1['BTC Notional']=(1000/d1['Traded Price'])
d1.head(100)

Traded Price  Traded Size  Traded Notional      Timestamp  BTC Notional
0       43160.00      0.00056        24.169600  1646092800000      0.023170
1       43160.01      0.00052        22.443205  1646092800003      0.023170
2       43160.01      0.00182        78.551218  1646092800004      0.023170
3       43160.01      0.00631       272.339663  1646092800005      0.023170
4       43160.00      0.00229        98.836400  1646092800006      0.023170
..           ...          ...              ...            ...           ...
95      43175.38      0.07999      3453.598646  1646092801258      0.023161
96      43176.16      0.05000      2158.808000  1646092801258      0.023161
97      43176.16      0.01158       499.979933  1646092801258      0.023161
98      43176.46      0.29102     12565.213389  1646092801258      0.023161
99      43172.07      0.10000      4317.207000  1646092801272      0.023163

[100 rows x 5 columns]

Starting at trade price of 43160.00
1. First the algo needs to buy 0.023170 BTC (1000USD equivalent) position B1.
2. Now, if the trade price goes down by 1USDt to 43159, it should buy another 1000USD equivalent BTC, B2.
3. Now, goes to 43161, buy another 1000USD, B3. Or if goes to 43158, buy another 1000USD, B4. And if it goes to 43157 or 43162, buy another 1000 USD, B5. And now as it as 5k USD in buy, it should stop buying anymore.
4. If the trade price goes to 43163, 3 points for B1, so it should sell the 0.023170BTC and get 0.023170 x 43163= 1000.08, i.e 8 cents profit, position S1. And if to 43162, 3 points for B2, so should sell again, S3.